dfget/core/uploader/uploader.go

```diff
@@ -120,27 +120,38 @@ func launch(cfg *config.Config, p2pPtr *unsafe.Pointer) error {
 	return fmt.Errorf("start peer server error and retried at most %d times", retryCount)
 }
 
-func waitForStartup(result chan error, p2pPtr *unsafe.Pointer) error {
-	select {
-	case err := <-result:
-		tmp := loadSrvPtr(p2pPtr)
-		if err == nil {
-			logrus.Infof("reuse exist server on port:%d", tmp.port)
-			tmp.setFinished()
-		}
-		return err
-	case <-time.After(100 * time.Millisecond):
-		// The peer server go routine will block and serve if it starts successfully.
-		// So we have to wait a moment and check again whether the peer server is
-		// started.
-		tmp := loadSrvPtr(p2pPtr)
-		if tmp == nil {
-			return fmt.Errorf("initialize peer server error")
-		}
-		if !uploaderAPI.PingServer(tmp.host, tmp.port) {
-			return fmt.Errorf("can't ping port:%d", tmp.port)
+func waitForStartup(result chan error, p2pPtr *unsafe.Pointer) (err error) {
+	ticker := time.NewTicker(5 * time.Millisecond)
+	defer ticker.Stop()
+	timeout := time.After(233 * time.Millisecond)
```

Comment:
why to use this value `233 * time.Millisecond`. How about declaring it a const value?

Reply:
It's not a multiple of `5`, is a prime number. And it's enough for retry to start a http server. 😄

Reply:
In fact, it's weird for 233ms. Maybe other contributors will feel the same way. Could you please add a comment for it? In addition, how about declaring it a const value as @allencloud said?

Reply:
> why is 233 ms, not others?

It's weird for us because of unfamiliarity, but it's really the one. There are many things in the world that are unexpected. I have already commented on this function to describe why this number jumps out by itself. It jumps into my brain, and I have to write it down.

> why not declare it as a const value

I think that maybe it's unnecessary to declare it as a const value. The code here can explain its meaning by a very simple and clear way by itself: it's the timeout of waiting for starting server. Just like the variable ticker with `5ms` above. It will lose its meaning when it appears in any other places. And the important thing is that it's very hard to come up with a name of the constant value which can explain itself without context.

Reply:
Oh, Man!